### PR TITLE
Transaction forwarding limit

### PR DESCRIPTION
**Pull request by sakridge**
#### Problem

Unlimited packet forwarding

#### Summary of Changes

Limit packets forwarded

Fixes #

### CODE REVIEW

**Comment by pyk** [comment]
@sakridge is this changes will greatly affect the TPS ser? 

thank you

**Comment by 0xSeaj** [comment]
@sakridge probably a dumb question, but as someone who doesn't have a ton of experience at this level, I'd like to know how you come up with the limit number?

File: core/src/banking_stage.rs [hunk 1]
```diff
@@ -315,11 +319,21 @@ impl BankingStage {
...
+        const MAX_BYTES_PER_INTERVAL: usize = MAX_BYTES_PER_SECOND * INTERVAL_MS as usize / 1000;
+        const MAX_BYTES_BUDGET: usize = MAX_BYTES_PER_INTERVAL * 5;
+        data_budget.update(INTERVAL_MS, |bytes| {
+            std::cmp::min(bytes + MAX_BYTES_PER_INTERVAL, MAX_BYTES_BUDGET)
+        });
         for p in packets {
-            socket.send_to(&p.data[..p.meta.size], &tpu_forwards)?;
+            if data_budget.take(p.meta.size) {
+                socket.send_to(&p.data[..p.meta.size], &tpu_forwards)?;
+            }
```
Comment (thesoftwarejedi) [inline core/src/banking_stage.rs:336]: Should have `else break;`?

**Comment by sagivo** [comment]
did anyone review and approve this PR? Just curios how we merge code that no one approved.

File: core/src/banking_stage.rs [hunk 2]
```diff
@@ -315,11 +319,21 @@ impl BankingStage {
         socket: &std::net::UdpSocket,
         tpu_forwards: &std::net::SocketAddr,
         unprocessed_packets: &UnprocessedPackets,
+        data_budget: &DataBudget,
     ) -> std::io::Result<()> {
         let packets = Self::filter_valid_packets_for_forwarding(unprocessed_packets.iter());
         inc_new_counter_info!("banking_stage-forwarded_packets", packets.len());
+        const INTERVAL_MS: u64 = 100;
+        const MAX_BYTES_PER_SECOND: usize = 10_000 * 1200;
```
Comment (sagivo) [inline core/src/banking_stage.rs:327]: any reason for these magic numbers?

**Pull request merged**